### PR TITLE
Migrate build files to npm package

### PR DESCRIPTION
**Pull request by retlehs**
moved to https://github.com/roots/sage-webpack-utils

### CODE REVIEW

File: assets/build/webpack.plugin.browsersync.js [hunk 1]
```diff
@@ -1,61 +0,0 @@
...
-    this.compiler = compiler;
-    compiler.plugin('done', () => {
-      if (!this.watcher) {
-        this.watcher = browserSync.create();
-        compiler.plugin('compilation', () => this.watcher.notify('Rebuilding...'));
-        this.start();
-      }
-      /* You may optionally add custom logic here to trigger either of the following */
-      // this.watcher.reload()
-      // this.watcher.reload({ stream: true })
```
Comment (oxyc) [inline assets/build/webpack.plugin.browsersync.js]: Any chance you could expose this somehow?

**Comment by ciromattia** [comment]
Uhm... I don't know if it's the correct place to ask, but I usually tweak the build files accordingly to the project, e.g. added postcssInlineSvg() to scss plugin loader postcss or 
Where am I supposed to put such tweaks, after this change?

**Comment by retlehs** [comment]
this isn't ready yet, you'll still be able to make the config changes you need from this repo

**Comment by eshimischi** [comment]
Why did you decide to move the build files into the npm package?

**Comment by mAAdhaTTah** [comment]
The eslintrc file should probably stay, as an IDE generally won't read it from a subpackage. Best practice for that tends to be the use of `extends`. Maybe an `eslint-preset-sage` would be appropriate?

**Comment by swalkinshaw** [comment]
@eshimischi there's a bunch of reasons:

* keep build related files out of the deployed theme
* separation of concerns: it's all JS and should be managed as a JS/npm package
* other themes/projects could more easily use it
* keep the theme root directory limited to actual theme related files
* easier to refactor build process

**Comment by eshimischi** [comment]
@swalkinshaw thanks for the clarifications. I do understand.

**Comment by eshimischi** [comment]
@swalkinshaw i'm working on my wordpress theme that based on Sage and also would love to make sage-rollup-utils package that implement rollup.js instead of wordpress 2

**Comment by retlehs** [comment]
@eshimischi yep, this isn't ready yet — and also not ready for testing.

**Pull request closed without merging**